pkg/service/roomservice.go

```diff
@@ -215,6 +215,36 @@ func (s *RoomService) SendData(ctx context.Context, req *livekit.SendDataRequest
 	return &livekit.SendDataResponse{}, nil
 }
 
+func (s *RoomService) UpdateRoomMetadata(ctx context.Context, req *livekit.UpdateRoomMetadataRequest) (*livekit.Room, error) {
+	if err := EnsureAdminPermission(ctx, req.Room); err != nil {
+		return nil, twirpAuthError(err)
+	}
+
+	room, err := s.roomStore.LoadRoom(ctx, req.Room)
+	if err != nil {
+		return nil, err
+	}
+
+	participants, err := s.roomStore.ListParticipants(ctx, req.Room)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(participants) > 0 {
+		err := s.writeMessage(ctx, req.Room, participants[0].Identity, &livekit.RTCNodeMessage{
+			Message: &livekit.RTCNodeMessage_UpdateRoomMetadata{
+				UpdateRoomMetadata: req,
+			},
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	room.Metadata = req.Metadata
```

Comment:
we should persist the room metadata here as well in case no participants have joined.

in the future we'll fix the remote messaging protocol to not require a participant. it's currently kind of hacky. (unrelated to your implementation)

Reply:
Should I send room update when a participant becomes active ( so the metadata is synchronized ).
In this [line](https://github.com/livekit/livekit-server/blob/cdb04248fb0af7c167d7c168142597d5d831d524/pkg/rtc/room.go#L177) ?

Reply:
You don't need to. JoinResponse contains the Room object, and therefore metadata.